### PR TITLE
HBASE-28676 Make pre commit build for 3.x to only run jdk 17 test

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  0s |  master passed  |
| +1 :green_heart: |  compile  |   1m 58s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 31s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   2m  0s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 49s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 58s |  the patch passed  |
| -0 :warning: |  javac  |   1m 58s |  root generated 75 new + 0 unchanged - 0 fixed = 75 total (was 0)  |
| +1 :green_heart: |  shadedjars  |   5m 24s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 53s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   1m 39s |  root in the patch failed.  |
|  |   |  28m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6001 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 0fdbc457f351 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 046e9d5540 |
| Default Java | Eclipse Adoptium-17.0.10+7 |
| javac | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/1/artifact/yetus-jdk17-hadoop3-check/output/diff-compile-javac-root.txt |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/1/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/1/testReport/ |
| Max. process+thread count | 300 (vs. ulimit of 30000) |
| modules | C: hbase-build-configuration . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 26s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 46s |  master passed  |
| -1 :x: |  compile  |   0m 16s |  root in master failed.  |
| +1 :green_heart: |  spotless  |   0m 56s |  branch has no errors when running spotless:check.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 18s |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 27s |  the patch passed  |
| -0 :warning: |  javac  |   8m 27s |  root generated 1206 new + 0 unchanged - 0 fixed = 1206 total (was 0)  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  xml  |   0m  1s |  The patch has 2 ill-formed XML file(s).  |
| +1 :green_heart: |  hadoopcheck  |   5m  7s |  Patch does not cause any errors with Hadoop 3.3.6.  |
| +1 :green_heart: |  spotless  |   0m 41s |  patch has no errors when running spotless:check.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 18s |  The patch does not generate ASF License warnings.  |
|  |   |  30m 22s |   |


| Reason | Tests |
|-------:|:------|
| XML | Parsing Error(s): |
|   | hbase-build-configuration/pom.xml |
|   | pom.xml |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6001 |
| Optional Tests | dupname asflicense spotless shellcheck shelldocs javac hadoopcheck xml compile |
| uname | Linux a27884dbf7a4 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 046e9d5540 |
| Default Java | Eclipse Adoptium-17.0.10+7 |
| compile | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/1/artifact/yetus-general-check/output/branch-compile-root.txt |
| javac | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/1/artifact/yetus-general-check/output/diff-compile-javac-root.txt |
| xml | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/1/artifact/yetus-general-check/output/xml.txt |
| Max. process+thread count | 189 (vs. ulimit of 30000) |
| modules | C: hbase-build-configuration . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/1/console |
| versions | git=2.34.1 maven=3.8.6 shellcheck=0.8.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  2s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 53s |  master passed  |
| -1 :x: |  compile  |   0m 15s |  root in master failed.  |
| +1 :green_heart: |  spotless  |   0m 44s |  branch has no errors when running spotless:check.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 48s |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 42s |  the patch passed  |
| -0 :warning: |  javac  |   7m 42s |  root generated 1179 new + 0 unchanged - 0 fixed = 1179 total (was 0)  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  xml  |   0m  1s |  The patch has 2 ill-formed XML file(s).  |
| +1 :green_heart: |  hadoopcheck  |   4m 56s |  Patch does not cause any errors with Hadoop 3.3.6.  |
| +1 :green_heart: |  spotless  |   0m 41s |  patch has no errors when running spotless:check.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 22s |  The patch does not generate ASF License warnings.  |
|  |   |  28m  4s |   |


| Reason | Tests |
|-------:|:------|
| XML | Parsing Error(s): |
|   | hbase-build-configuration/pom.xml |
|   | pom.xml |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6001 |
| Optional Tests | dupname asflicense spotless shellcheck shelldocs javac hadoopcheck xml compile |
| uname | Linux fe43716f7415 5.4.0-174-generic #193-Ubuntu SMP Thu Mar 7 14:29:28 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / ff1e52bfce |
| Default Java | Eclipse Adoptium-17.0.10+7 |
| compile | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/2/artifact/yetus-general-check/output/branch-compile-root.txt |
| javac | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/2/artifact/yetus-general-check/output/diff-compile-javac-root.txt |
| xml | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/2/artifact/yetus-general-check/output/xml.txt |
| Max. process+thread count | 185 (vs. ulimit of 30000) |
| modules | C: hbase-build-configuration . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/2/console |
| versions | git=2.34.1 maven=3.8.6 shellcheck=0.8.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
Filed HBASE-28679 for the xml problem, and we can not move releaseTarget to 17 because there are tests failures on mockito, which may require code change. Plan to solve it in HBASE-28675.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 47s |  master passed  |
| +1 :green_heart: |  compile  |   1m 57s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 25s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 56s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 58s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 58s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 26s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 56s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 268m 26s |  root in the patch passed.  |
|  |   | 299m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/2/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6001 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 100ee231074d 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / ff1e52bfce |
| Default Java | Eclipse Adoptium-17.0.10+7 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/2/testReport/ |
| Max. process+thread count | 8957 (vs. ulimit of 30000) |
| modules | C: hbase-build-configuration . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   3m 48s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --blanks-eol-ignore-list --blanks-tabs-ignore-list  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 52s |  |  master passed  |
| -1 :x: |  compile  |   0m 15s | [/branch-compile-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/3/artifact/yetus-general-check/output/branch-compile-root.txt) |  root in master failed.  |
| +1 :green_heart: |  spotless  |   0m 42s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 46s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 35s |  |  the patch passed  |
| -0 :warning: |  javac  |   7m 35s | [/results-compile-javac-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/3/artifact/yetus-general-check/output/results-compile-javac-root.txt) |  root generated 1179 new + 0 unchanged - 0 fixed = 1179 total (was 0)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  hadoopcheck  |   4m 57s |  |  Patch does not cause any errors with Hadoop 3.3.6.  |
| +1 :green_heart: |  spotless  |   0m 42s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 17s |  |  The patch does not generate ASF License warnings.  |
|  |   |  30m 45s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6001 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs spotless javac xmllint hadoopcheck compile |
| uname | Linux e48bb3803d5c 5.4.0-174-generic #193-Ubuntu SMP Thu Mar 7 14:29:28 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9ad500af5bfb9573da1dc9520010ab23dcf3ebe5 |
| Default Java | Eclipse Adoptium-17.0.10+7 |
| Max. process+thread count | 187 (vs. ulimit of 30000) |
| modules | C: hbase-build-configuration . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/3/console |
| versions | git=2.34.1 maven=3.8.6 shellcheck=0.8.0 xmllint=20913 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
The error for compile for the current master branch is expected, since we change to use jdk17 for compiling, we need to add some compile args for error prone.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 27s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-list --blanks-tabs-ignore-list --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  9s |  |  master passed  |
| +1 :green_heart: |  compile  |   1m 58s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 53s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 37s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 50s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 56s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 56s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 53s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 28s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 332m 12s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/3/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   | 412m 39s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/3/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6001 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 2a7ec825152b 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9ad500af5bfb9573da1dc9520010ab23dcf3ebe5 |
| Default Java | Eclipse Adoptium-17.0.10+7 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/3/testReport/ |
| Max. process+thread count | 7336 (vs. ulimit of 30000) |
| modules | C: hbase-build-configuration . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/3/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 47s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 34s |  |  master passed  |
| -1 :x: |  compile  |   0m 24s | [/branch-compile-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/4/artifact/yetus-general-check/output/branch-compile-root.txt) |  root in master failed.  |
| +1 :green_heart: |  spotless  |   0m 58s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 44s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   9m 57s |  |  the patch passed  |
| -0 :warning: |  javac  |   9m 57s | [/results-compile-javac-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/4/artifact/yetus-general-check/output/results-compile-javac-root.txt) |  root generated 1179 new + 0 unchanged - 0 fixed = 1179 total (was 0)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  hadoopcheck  |   6m 41s |  |  Patch does not cause any errors with Hadoop 3.3.6.  |
| +1 :green_heart: |  spotless  |   0m 57s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 25s |  |  The patch does not generate ASF License warnings.  |
|  |   |  37m 36s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6001 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs spotless javac xmllint hadoopcheck compile |
| uname | Linux 8fc13d8fa23a 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 407e80f1bea0a7d1f5e29069fd4d6b9ce4b8f79c |
| Default Java | Eclipse Adoptium-17.0.10+7 |
| Max. process+thread count | 186 (vs. ulimit of 30000) |
| modules | C: hbase-build-configuration . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/4/console |
| versions | git=2.34.1 maven=3.8.6 shellcheck=0.8.0 xmllint=20913 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 42s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 41s |  |  master passed  |
| +1 :green_heart: |  compile  |   2m 48s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 29s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 38s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 55s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 47s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 47s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   3m 18s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 53s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 311m 26s |  |  root in the patch passed.  |
|  |   | 351m 13s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/4/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6001 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 741cec0e2f13 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 407e80f1bea0a7d1f5e29069fd4d6b9ce4b8f79c |
| Default Java | Eclipse Adoptium-17.0.10+7 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/4/testReport/ |
| Max. process+thread count | 8231 (vs. ulimit of 30000) |
| modules | C: hbase-build-configuration . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6001/4/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
OK, good. Xml check can pass now.

**Comment by Apache9** [comment]
Merging this will cause the nightly build for master and branch-3 to fail since we still use jdk11 for general checks.

Will start to work on HBASE-28678 ASAP after merging this PR.

File: pom.xml [hunk 1]
```diff
@@ -809,7 +809,7 @@
     <tar.name>${project.build.finalName}.tar.gz</tar.name>
     <maven.build.timestamp.format>yyyy-MM-dd'T'HH:mm</maven.build.timestamp.format>
     <buildDate>${maven.build.timestamp}</buildDate>
-    <compileSource>1.8</compileSource>
```
Comment (ndimiduk) [inline pom.xml:812]: Big jump!

File: pom.xml [hunk 2]
```diff
@@ -809,7 +809,7 @@
...
+    <compileSource>17</compileSource>
     <releaseTarget>8</releaseTarget>
```
Comment (ndimiduk) [inline pom.xml:813]: What does it mean to have compile target at 17 but release target at 8? Can javac emit bytecode for 17 features for an 8 VM?

We are dropping JDK8 runtime support, so I think we can move the release target up as well.

**Review by ndimiduk — APPROVED** [review]

**Comment by Apache9** [comment]
> Filed [HBASE-28679](https://issues.apache.org/jira/browse/HBASE-28679) for the xml problem, and we can not move releaseTarget to 17 because there are tests failures on mockito, which may require code change. Plan to solve it in [HBASE-28675](https://issues.apache.org/jira/browse/HBASE-28675).

@ndimiduk See here, I tried to jump to 17 too for releaseTarget at first, but it breaks mockito, so I plan to address it in HBASE-28675 since it may require java code changes.

File: pom.xml (hunk 2 above)
Comment (Apache9) [reply pom.xml:813]: ReleaseTarget 17 will break mockito, need to dig more...

**Comment by ndimiduk** [comment]
Makes sense. Thanks @Apache9 !

**Pull request merged**